questiontype.php

```diff
@@ -457,7 +487,14 @@ public function import_from_xml($data, $question, qformat_xml $format, $extra =
         $rowsxml = $data['#']['row'];
 
         foreach ($rowsxml as $rowxml) {
-            $question->rows_shorttext[$index] = $format->getpath($rowxml, ['#', 'shorttext', 0, '#'], '');
+            // Support both legacy: <shorttext>text</shorttext>
+            // and new: <shorttext format="..."><text>html</text></shorttext>
+            $shorttexthtml = $format->getpath($rowxml, ['#', 'shorttext', 0, '#', 'text', 0, '#'], null);
```

Comment:
There exists a function called import_text() that does some small post-processing. You should probably use that, so:

```
$shorttexthtml = $format->getpath($rowxml, ['#', 'shorttext', 0, '#', 'text'], null);
if ($shorttexthtml !== null) {
   // Postprocess text node
   $shorttexthtml = import_text($shorttexthtml);
} else {
   // An older exported matrix question
   $shorttexthtml = $format->getpath($rowxml, ['#', 'shorttext', 0, '#'], '');
}
```